proto/http.go

```diff
@@ -0,0 +1,158 @@
+package proto
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/opentracing/opentracing-go"
+	otlog "github.com/opentracing/opentracing-go/log"
+)
+
+// CompressionType for encoding and decoding requests and responses.
+type CompressionType int
+
+// Values for CompressionType
+const (
+	NoCompression CompressionType = iota
+	RawSnappy
+)
+
+const messageSizeLargerErrFmt = "received message larger than max (%d vs %d)"
+
+// ParseProtoReader parses a compressed proto from an io.Reader.
+func ParseProtoReader(ctx context.Context, reader io.Reader, expectedSize, maxSize int, req proto.Message, compression CompressionType) error {
```

Comment:
Are functions in this file related to moving ring to dskit? I don't think so. If not, I would remove them from this PR and discuss them separately, so we can get the ring merge merged sooner.